Mathlib/AlgebraicTopology/SimplicialSet/HomotopyCat.lean

```diff
@@ -0,0 +1,456 @@
+first restricting from simplicial sets to 2-truncated simplicial sets (throwing away the data that
+is not used for the construction of the homotopy category) and then composing with an analogously
+defined `SSet.hoFunctor₂ : SSet.Truncated.{u} 2 ⥤ Cat.{u,u}` implemented relative to the syntax of
+the 2-truncated simplex category.
+
+It should be relatively straightforward to show that these constructions agree:
+
+def hoFunctor.ofTwoTruncation.iso (V : SSet) :
+    SSet.hoFunctor₂Obj ((truncation 2).obj V) ≅ SSet.hoCat V := sorry
+
+def hoFunctor.ofTwoTruncation.natIso :
+     truncation 2 ⋙ SSet.hoFunctor₂ ≅ SSet.hoFunctor' := sorry
+
+but we leave this for future work.
+
+The functor `SSet.hoFunctor` is shown to be left adjoint to the nerve by providing an analogous
+decomposition of the nerve functor and then composing a pair of adjunctions, which factor through
+the category of 2-truncated simplicial sets.
+-/
+
+namespace CategoryTheory
+open Category Limits Functor Opposite Simplicial
+universe v u
+
+section
+
+/-- A simplicial set `S` has an underlying refl quiver with `S _[0]` as its underlying type.-/
+def SSet.OneTruncation (S : SSet) := S _[0]
+
+/-- The source vertex of `f : S _[1]` for use in defining the underlying refl quiver.-/
+def SSet.OneTruncation.src {S : SSet} (f : S _[1]) : OneTruncation S := S.δ 1 f
+
+/-- The target vertex of `f : S _[1]` for use in defining the underlying refl quiver.-/
+def SSet.OneTruncation.tgt {S : SSet} (f : S _[1]) : OneTruncation S := S.δ 0 f
+
+/-- The hom-types of the refl quiver underlying a simplicial set `S` are subtypes of `S _[1]`.-/
+def SSet.OneTruncation.Hom {S : SSet} (X Y : OneTruncation S) :=
+  {p : S _[1] // src p = X ∧ tgt p = Y}
+
+/-- A simplicial set `S` has an underlying refl quiver `SSet.OneTruncation S`.-/
+instance (S : SSet) : ReflQuiver (SSet.OneTruncation S) where
+  Hom X Y := SSet.OneTruncation.Hom X Y
+  id X := by
+    refine ⟨S.σ 0 X, ?_, ?_⟩ <;> change (S.δ _ (S.σ _ _)) = X
+    · apply SSet.δ_comp_σ_succ_apply 0
+    · apply SSet.δ_comp_σ_self_apply 0
+
+/-- The functor that carries a simplicial set to its underlying refl quiver.-/
+def SSet.oneTruncation : SSet.{u} ⥤ ReflQuiv.{u,u} where
+  obj S := ReflQuiv.of (OneTruncation S)
+  map {S T} F := {
+    obj := F.app (op [0])
+    map := fun f => by
+      refine ⟨F.app (op [1]) f.1, ?_, ?_⟩
+      · change (F.app _ ≫ _) _ = _
+        rw [← F.naturality]
+        exact congrArg (F.app _) f.2.1
+      · change (F.app _ ≫ _) _ = _
+        rw [← F.naturality]
+        exact congrArg (F.app _) f.2.2
+    map_id := fun X => by
+      change ({..} : Subtype _) = {..}
+      congr
+      change _ = (F.app _ ≫ _) _
+      rw [← F.naturality]
+      rfl
+  }
+
+@[ext] lemma hom_ext {S : SSet} {x y : SSet.OneTruncation S} {f g : x ⟶ y} :
+    f.1 = g.1 → f = g := Subtype.ext
+
+section
+variable {C : Type u} [Category.{v} C]
+
+theorem opstuff.{w} (V : Cᵒᵖ ⥤ Type w) {X Y Z : C} {α : X ⟶ Y} {β : Y ⟶ Z} {γ : X ⟶ Z} {φ} :
```

Comment:
Is this name intended to stay? Or is this WIP?

Reply:
@digama0 what should this have been called?

Reply:
I think it should be a `private lemma`. Perhaps `map_map_of_eq`?

Reply:
Done.